File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/order/domain/PaymentInfo.java
package band.gosrock.domain.domains.order.domain;


import band.gosrock.domain.common.vo.Money;
import javax.persistence.Embeddable;
import javax.persistence.Embedded;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class PaymentInfo {

    // 총 결제금액
    @Embedded private Money paymentAmount;

    // 공급가액
    @Embedded private Money supplyAmount;

    // 총 할인 금액
    @Embedded private Money discountAmount;
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/order/domain/PaymentInfo.kt
package band.gosrock.domain.domains.order.domain

import band.gosrock.domain.common.vo.Money
import javax.persistence.AttributeOverride
import javax.persistence.Column
import javax.persistence.Embeddable
import javax.persistence.Embedded

@Embeddable
class PaymentInfo() {

    @Embedded
    @AttributeOverride(name = "amount", column = Column(name = "payment_amount"))
    var paymentAmount: Money? = null
        protected set

    @Embedded
    @AttributeOverride(name = "amount", column = Column(name = "supply_amount"))
    var supplyAmount: Money? = null
        protected set

    @Embedded
    @AttributeOverride(name = "amount", column = Column(name = "discount_amount"))
    var discountAmount: Money? = null
        protected set

    companion object {
        @JvmStatic
        fun of(paymentAmount: Money, supplyAmount: Money, discountAmount: Money): PaymentInfo =
            PaymentInfo().apply {
                this.paymentAmount = paymentAmount
                this.supplyAmount = supplyAmount
                this.discountAmount = discountAmount
            }
    }
}
